vpr/SRC/route/route_timing.c

```diff
@@ -556,35 +686,33 @@ bool timing_driven_route_net(int inet, int itry, float pres_fac, float max_criti
 			pin_criticality[ipin] = pow(pin_criticality[ipin], criticality_exp);
 			
 			/* Cut off pin criticality at max_criticality. */
-			pin_criticality[ipin] = min(pin_criticality[ipin], max_criticality);
+			pin_criticality[ipin] = min(pin_criticality[ipin], max_criticality);			
 		}
 	}
 
-	num_sinks = g_clbs_nlist.net[inet].num_sinks();
-	heapsort(sink_order, pin_criticality, num_sinks, 0);
-
+	// compare the criticality of different sink nodes
+	sort(begin(remaining_targets), end(remaining_targets), Criticality_comp{pin_criticality});
 	/* Update base costs according to fanout and criticality rules */
 
-	largest_criticality = pin_criticality[sink_order[1]];
+
+	float largest_criticality = pin_criticality[remaining_targets[0]];
 	update_rr_base_costs(inet, largest_criticality);
+	
 
-	mark_ends(inet); /* Only needed to check for multiply-connected SINKs */
+	int itarget {0};	// not sure if necessary...
+	// explore in order of decreasing criticality (no longer need sink_order array)
+	for (int target_pin : remaining_targets) {

```

Comment:
I tend to avoid the range based for loop if I need to also carry around an index. It is more straight forward and less error prone to use the standard for loop